### PR TITLE
WINC-1001: [wmco] Use standard library errors package

### PR DESCRIPTION
**Pull request by alinaryan**
Uses the standard library errors package given the package, github.com/pkg/errors, has been archived.

### CODE REVIEW

File: test/e2e/logs_test.go
Comment (saifshaikh48) [inline test/e2e/logs_test.go:76]: I don't think this is needed, please undo

**Review by saifshaikh48 — COMMENTED** [review]
Looks good, have a few comments

**Comment by sebsoto** [comment]
I've validated that this correctly removes all imports of `github.com/pkg/errors`

File: controllers/windowsmachine_controller.go
Comment (sebsoto) [inline controllers/windowsmachine_controller.go:79]: Missing wrap here

File: controllers/windowsmachine_controller.go
Comment (sebsoto) [inline controllers/windowsmachine_controller.go:84]: Missing wrap here

File: controllers/windowsmachine_controller.go
Comment (sebsoto) [inline controllers/windowsmachine_controller.go:90]: Missing wrap here

File: pkg/cluster/config.go
Comment (sebsoto) [inline pkg/cluster/config.go:207]: theres no error to wrap here, please remove

File: pkg/csr/csr.go
Comment (sebsoto) [inline pkg/csr/csr.go:202]: should not wrap here

File: pkg/daemon/controller/controller.go
Comment (sebsoto) [inline pkg/daemon/controller/controller.go:388]: Shouldnt wrap here

File: pkg/daemon/fake/manager.go
Comment (sebsoto) [inline pkg/daemon/fake/manager.go:84]: Shouldnt wrap here

File: pkg/daemon/fake/manager.go
Comment (sebsoto) [inline pkg/daemon/fake/manager.go:111]: this change is unecessary

File: pkg/daemon/fake/manager.go
Comment (sebsoto) [inline pkg/daemon/fake/manager.go:143]: Unecessary

File: pkg/daemon/fake/manager.go
Comment (sebsoto) [inline pkg/daemon/fake/manager.go:152]: Unecessary

File: pkg/daemon/fake/manager.go
Comment (sebsoto) [inline pkg/daemon/fake/manager.go:163]: Unecessary

File: pkg/daemon/fake/winsvc.go
Comment (sebsoto) [inline pkg/daemon/fake/winsvc.go:23]: Shouldnt wrap here

File: pkg/daemon/fake/winsvc.go
Comment (sebsoto) [inline pkg/daemon/fake/winsvc.go:29]: Shouldnt wrap here

File: pkg/daemon/fake/winsvc.go
Comment (sebsoto) [inline pkg/daemon/fake/winsvc.go:55]: Shouldnt wrap here

File: pkg/daemon/fake/winsvc.go
Comment (sebsoto) [inline pkg/daemon/fake/winsvc.go:79]: Shouldnt wrap here

**Review by sebsoto — CHANGES_REQUESTED** [review]
Pausing this review.
Please address all current comments and ensure you are not attempting to reference an error which may or may not exist when replacing errors.New().

File: pkg/daemon/cleanup/cleanup.go
Comment (saifshaikh48) [inline pkg/daemon/cleanup/cleanup.go:1]: I don't see fmt being imported by the WICD files (`daemon` pkg). Are you able to `make build-daemon` to make sure it compiles?

File: test/e2e/secrets_test.go
Comment (saifshaikh48) [inline test/e2e/secrets_test.go:119]: I don't think this error needs to be wrapped. `err` will always hold `nil` value here (if it's non-nil it is returned before entering the loop

File: pkg/wiparser/wiparser.go
Comment (saifshaikh48) [inline pkg/wiparser/wiparser.go:27]: nit `: %w` instead of `, %w`

File: pkg/cluster/config.go
Comment (saifshaikh48) [inline pkg/cluster/config.go:274]:
```suggestion
	if err := ValidateCIDR(serviceCIDR); err != nil {
		return "", fmt.Errorf("invalid cluster service CIDR: %w", err)
```

File: test/e2e/logs_test.go
Comment (aravindhp) [inline test/e2e/logs_test.go:68]: Should this be a separate PR so that we can backport it easily? i.e. fix this file separately. The log collection flake is showing up quite often and I would like us to get to the bottom of that. cc: @sebsoto

Comment (sebsoto) [reply test/e2e/logs_test.go:68]: yes lets break this file's changes out into its own PR, otherwise the process to get this fix in to previous branches will take way too long

Comment (alinaryan) [reply test/e2e/logs_test.go:68]: Sounds good, I'll break them up ASAP

File: pkg/daemon/cleanup/cleanup.go
Comment (alinaryan) [reply pkg/daemon/cleanup/cleanup.go:1]: Thanks for catching this. "fmt" and "errors" were missing from a few files - I added them in and make build-daemon compiled successfully

Comment (alinaryan) [reply pkg/daemon/cleanup/cleanup.go:1]: I also went through every file and ensured fmt was being imported

**Review by saifshaikh48 — COMMENTED** [review]
/lgtm

Thanks for powering through this

**Review by sebsoto — APPROVED** [review]
Great work on this

**Comment by saifshaikh48** [comment]
@alinaryan please investigate the test failures

Separately, some of the error messages are ending with `%!w(<nil>)` as the wrapped error
```
"error": "error configuring host with address e2e-nfqjg: error getting node object: unable to find node with address 10.0.128.7: %!w(<nil>)" 
"error": "error updating endpoints object with list of endpoint addresses: unable to sync metrics endpoints: %!w(<nil>)"
```

**Comment by sebsoto** [comment]
> Separately, some of the error messages are ending with %!w(<nil>) as the wrapped error

Theres a good chance this is because errors.Wrap was being called when the error was nil, you need to add if err != nil in that case.

errors.Wrap would return nil if err was nil

**Comment by alinaryan** [comment]
@saifshaikh48 Can you link where you see that error? I searched for it but I can't find that in any of the test build logs/prow links

**Comment by saifshaikh48** [comment]
@alinaryan [WMCO logs](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_windows-machine-config-operator/1512/pull-ci-openshift-windows-machine-config-operator-master-azure-e2e-operator/1641897888164352000/artifacts/azure-e2e-operator/gather-extra/artifacts/pods/openshift-windows-machine-config-operator_windows-machine-config-operator-5ccbcf7945-98rj6_manager.log) on all platforms

File: pkg/csr/csr.go
Comment (saifshaikh48) [reply pkg/csr/csr.go:84]: either works given validateCSRContents() only returns error when it also returns false

Comment (saifshaikh48) [reply pkg/csr/csr.go:84]: Just replacing wouldn't result in the desired behavior. You can join the conditions
```
	if valid, err := a.validateCSRContents(); !valid && err != nil {
		// if the validation fails and error returned is nil, returns nil
		return fmt.Errorf("could not validate contents for approval of CSR: %s: %w", a.csr.Name, err)
	}
```
I'd like @mansikulkarni96's eyes on this too, might be out of scope of this PR

**Comment by saifshaikh48** [comment]
LGTM but nodes are not being bootstrapped. Looks like WICD cleanup is failing, [WMCO logs](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_windows-machine-config-operator/1512/pull-ci-openshift-windows-machine-config-operator-master-aws-e2e-operator/1643346195185143808/artifacts/aws-e2e-operator/gather-extra/artifacts/pods/openshift-windows-machine-config-operator_windows-machine-config-operator-5f8b99668d-5tcvv_manager.log)

File: pkg/csr/csr.go
Comment (mansikulkarni96) [reply pkg/csr/csr.go:84]: combining shouldn't be any issue, at least looking at the return values.

File: pkg/metadata/metadata.go
Comment (saifshaikh48) [inline pkg/metadata/metadata.go:133]: You can remove the `found` variable completely and just make this `err !=nil`

**Comment by saifshaikh48** [comment]
/lgtm
/hold

holding until approval is given again since there have been some changes since last applied

**Review by sebsoto — APPROVED** [review]

**Pull request merged**
